### PR TITLE
Added IsLightDismissed to PopupDismissedEvent

### LINKED ISSUE #1194: [Bug] UWP Popup dismiss triggers light dismiss
### Description

When closing a popup on UWP using the `Dismiss` method the `LightDismiss` method is also invoked.

### Steps to Reproduce

1. Open Xamarin.Community.Toolkit.Sample solution
2. Open the `PopupRenderer.uwp.cs` file and add a breakpoint at line 263: `Element.LightDismiss();`
3. Set the UWP project as the startup project
4. Run the sample app
5. Under **Views** > **Popup**, select **Popup with 1 Button** 
6. Tap the **OKAY** button to dismiss the popup

### Expected Behavior

The breakpoint on line 263 will **not** be hit and continuing execution will **not** invoke `LightDismiss`.

### Actual Behavior

The breakpoint on line 263 will be hit and continuing execution will invoke `LightDismiss`.

### Basic Information

- Version with issue: `1.1.0`
- Last known good version: `N/A`
- IDE: Visual Studio 2019 16.9.3
- Platform Target Frameworks:
  - UWP:  TargetVersion=19041, MinVersion=17763
- Nuget Packages:
  - Xamarin.CommunityToolkit 1.1.0
  - Xamarin.Forms 5.0.0.2012
  - Xamarin.Essentials.Interfaces 1.6.1
- Affected Devices:
  - UWP Desktop

### Workaround

None found

### Potential Solution

- Add an `isDismissing` flag
- Set to `true` before `Hide` is invoked
- Set to `false` after `Hide` is invoked
- Add an additional check in `OnClosing` to ensure that `Element.LightDismiss()` is only invoked when `isDismissing` is `false`

### LINKED ISSUE #1352: [Spec]  Add IsLightDismissed to Popup Events
# Add IsLightDismissed to Popup Events

Expose an `IsLightDismissed` property to the `PopupDismissedEventArgs` which will allow event listeners to understand if the popup was closed or if the user tapped in the area outside of the popup. 

*This was something that I included in the original implementation of XCT Popups that we are using internally and didn't include it as the code was migrated from Xamarin.Forms to XCT.*

# API

### Properties

| API | Description |
| ------------- | ------------- |
| IsLightDismissed | Gets or sets if the popup was dismissed via light dismiss. If true, then the user tapped outside the bounds of the popup (a light dismiss). If false, then the popup was dismissed by user action or via code. |

### Events
N/A

# Scenarios
This is a critical piece for anyone trying to build a popup library around the Xamarin Community Toolkit. We have an internal prism plugin that uses the original implementation of XCT Popups. It heavily depends on this property in the event listener mechanism. I am in the process of porting our internal library to open source and need this property exposed. Other projects that want to do similar things for other MVVM libraries will ultimately run into this problem.

# Platform Compatibility
- Target Frameworks: 
  - iOS:  ✅ 
  - Android: ✅ 
  - UWP:  ✅
 
# Backward Compatibility
There should be no need to make changes to the any dependencies as this is just exposing 1 new property.

# Difficulty : low

### PR DESCRIPTION
**Pull request by ahoefling**
### Description of Change ###
Adds `IsLightDismissed` property to the `PopupDismisedEventArgs`

Fires `Dismiss` event if the popup is closed via Light Dismiss.

### Bugs Fixed ###

- Fixes #1352 
- Fixes #1194

### API Changes ###

- [x] Has a linked Issue, and the Issue has been `approved`
- [ ] Has tests (if omitted, state reason in description)
- [ ] Has samples (if omitted, state reason in description)
- [x] Rebased on top of main at time of PR
- [x] Changes adhere to coding standard

- [ ] Updated [documentation](https://github.com/MicrosoftDocs/xamarin-communitytoolkit)

### CODE REVIEW

**Comment by ahoefling** [comment]
Updated PR to mark #1194 as fixed per confirmation from @chowarth 
- https://github.com/xamarin/XamarinCommunityToolkit/issues/1194#issuecomment-854456246

File: src/CommunityToolkit/Xamarin.CommunityToolkit/Views/Popup/UWP/PopupRenderer.uwp.cs [hunk 1]
```diff
@@ -252,14 +252,20 @@ SizeRequest IVisualElementRenderer.GetDesiredSize(double widthConstraint, double
...
 		UIElement? IVisualElementRenderer.GetNativeElement() => Control;
 
+		bool isOpen = true;
+
 		void OnDismissed(object? sender, PopupDismissedEventArgs e)
 		{
+			if (!isOpen)
+				return;
+
+			isOpen = false;
```
Comment (AndreiMisiukevich) [inline src/CommunityToolkit/Xamarin.CommunityToolkit/Views/Popup/UWP/PopupRenderer.uwp.cs:262]: Just a question: should we set ```isOpen = true``` somewhere? e.g. if I "reopen" the same popup?

File: src/CommunityToolkit/Xamarin.CommunityToolkit/Views/Popup/BasePopup.shared.cs [hunk 2]
```diff
@@ -151,17 +151,8 @@ public Size Size
...
-			// Note 1/9/2021
-			// Left empty by design
-			//
-			// This method needs to be left empty as it is not
-			// required for a child class to have an implementation
-			// of LightDismiss. This means if a renderer invokes
-			// this method nothing will happen unless the child
-			// class has an implementation.
-		}
+		protected internal virtual void LightDismiss() =>
```
Comment (AndreiMisiukevich) [inline src/CommunityToolkit/Xamarin.CommunityToolkit/Views/Popup/BasePopup.shared.cs:154]: ```OnLightDismissed``` ?

File: src/CommunityToolkit/Xamarin.CommunityToolkit/Views/Popup/UWP/PopupRenderer.uwp.cs (hunk 1 above)
Comment (ahoefling) [reply src/CommunityToolkit/Xamarin.CommunityToolkit/Views/Popup/UWP/PopupRenderer.uwp.cs:262]: Good feedback, I'll update this to set `isOpen = true` when it displays it on the screen

Comment (ahoefling) [reply src/CommunityToolkit/Xamarin.CommunityToolkit/Views/Popup/UWP/PopupRenderer.uwp.cs:270]: @AndreiMisiukevich I pushed some changes that set `isOpen = true` and I hooked into the UWP `Opened` event

**Comment by ahoefling** [comment]
I have pushed some more changes based on feedback, let me know if there is anything else I need to do

File: src/CommunityToolkit/Xamarin.CommunityToolkit/Views/Popup/UWP/PopupRenderer.uwp.cs [hunk 3]
```diff
@@ -283,6 +300,7 @@ protected virtual void Dispose(bool disposing)
 				Control = null;
 
 				Closed -= OnClosing;
```
Comment (AndreiMisiukevich) [inline src/CommunityToolkit/Xamarin.CommunityToolkit/Views/Popup/UWP/PopupRenderer.uwp.cs:302]: BTW, should it be ```Closing -= OnClosing```? I see that we do ```Closing += OnClosing``` in this file

Comment (AndreiMisiukevich) [inline src/CommunityToolkit/Xamarin.CommunityToolkit/Views/Popup/UWP/PopupRenderer.uwp.cs:302]:
```suggestion
				Closing -= OnClosing;
```

**Comment by jfversluis** [comment]
If we can finish up the last bits here then I think we're good to merge!

**Comment by ahoefling** [comment]
I thought I had this wrapped up but forgot to come back to it. I'll push a change in a little bit.

Thanks for the reminder

File: src/CommunityToolkit/Xamarin.CommunityToolkit/Views/Popup/UWP/PopupRenderer.uwp.cs (hunk 3 above)
Comment (ahoefling) [reply src/CommunityToolkit/Xamarin.CommunityToolkit/Views/Popup/UWP/PopupRenderer.uwp.cs:302]: `Closing` is an object from the UWP `Flyout` and can't be renamed. I can take the spirit of this change and rename `OnClosing -> OnClosed`
```c#
Closed -= OnClosed
```

Comment (ahoefling) [reply src/CommunityToolkit/Xamarin.CommunityToolkit/Views/Popup/UWP/PopupRenderer.uwp.cs:302]: This is a great catch as we would have a potential memory leak here.

Comment (ahoefling) [reply src/CommunityToolkit/Xamarin.CommunityToolkit/Views/Popup/UWP/PopupRenderer.uwp.cs:302]: This appears to the same feedback as the other thread.

**Comment by ahoefling** [comment]
I have rebased this branch with the latest changes on develop and made the requested changes. I believe we are all set, let me know if there is anything else needed.

**Review by jfversluis — APPROVED** [review]

**Pull request merged**